activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1233,16 +1233,32 @@ def offset!(value) # :nodoc:
       self
     end
 
-    # Specifies locking settings (default to +true+). For more information
+    # Specifies locking settings (defaults to `"FOR UPDATE"`). For more information
     # on locking, please see ActiveRecord::Locking.
-    def lock(locks = true)
-      spawn.lock!(locks)
+    def lock(lock = (lock_not_given = "FOR UPDATE"))
```

Comment:
Ref for default: https://github.com/rails/rails/blob/276b655f3606cf81b26464d135db9f24bacc1f0f/activerecord/lib/arel/select_manager.rb#L52-L63